### PR TITLE
Labels parameter

### PR DESCRIPTION
**Pull request by lcarva**
Overlapping commit from #422 needed to avoid unit testing conflicts. Merging this will also merge #422 .

### CODE REVIEW

File: tests/build/test_build_request.py [hunk 1]
```diff
@@ -1206,6 +1234,8 @@ def test_prod_is_custom_base_image(self, tmpdir, base_image, is_custom):
         build_request.set_params(**kwargs)
         build_json = build_request.render()
 
+        phase = 'prebuild_plugins'
+        plugin = 'add_labels_in_dockerfile'
```
Comment (mmilata) [inline tests/build/test_build_request.py]: These seem to be unused.

File: osbs/conf.py [hunk 2]
```diff
@@ -362,6 +363,15 @@ def get_build_imagestream(self):
     def get_proxy(self):
         return self._get_value("yum_proxy", self.conf_section, "yum_proxy")
 
+    def get_labels(self):
+        labels_list = self._get_value("labels", self.conf_section, "labels")
+        if labels_list is None:
+            return {}
+        # When read from config file, it needs to be deserialized
+        if type(labels_list) is not list:
+            labels_list = json.loads(labels_list)
```
Comment (mmilata) [inline osbs/conf.py]: Does this mean the config syntax is something like `labels = [("label1", "value1"), ("label2", "value2"), ("label3", "value3")]`?

Looks pretty weird to me:) But then again, we probably don't expect users to set labels in the configuration, do we? Should we just not allow setting this in the config?

File: tests/build/test_build_request.py [hunk 3]
```diff
@@ -1245,3 +1188,105 @@ def test_render_prod_request_with_koji_secret(self, tmpdir):
...
+        # Safe to call prior to build image being set
+        assert build_request.is_custom_base_image() is False
+
+        kwargs = get_sample_prod_params()
+        kwargs['base_image'] = base_image
+        build_request.set_params(**kwargs)
+        build_json = build_request.render()
+
+        phase = 'prebuild_plugins'
+        plugin = 'add_labels_in_dockerfile'
```
Comment (mmilata) [inline tests/build/test_build_request.py]: These seem to be unused.

**Comment by mmilata** [comment]
Two comments inline, otherwise LGTM.

File: osbs/conf.py (hunk 2 above)
Comment (lcarva) [inline osbs/conf.py]: Yes. It does look odd, but I don't suppose anyone would use it in the config. We could enhance the Configuration class to control the sources of certain item, but as it stands it does not support this behavior. Is it worth adding it?

File: tests/build/test_build_request.py (hunk 3 above)
Comment (lcarva) [inline tests/build/test_build_request.py]: Removed. Good catch!

File: osbs/conf.py (hunk 2 above)
Comment (mmilata) [inline osbs/conf.py]: I think we can live with this:)

**Comment by mmilata** [comment]
LGTM

**Comment by twaugh** [comment]
LGTM

**Pull request merged**